Anlab.Mvc/Controllers/AccountController.cs

```diff
@@ -290,6 +290,16 @@ public IActionResult ExternalLogin(string provider, string returnUrl = null)
                 return RedirectToAction(nameof(Login));
             }
 
+            if (info.LoginProvider.Equals("Google", StringComparison.OrdinalIgnoreCase))
+            {
+                var emailClaim = info.Principal.FindFirstValue(ClaimTypes.Email);
+                if (emailClaim.Contains("@ucdavis"))
+                {
+                    ErrorMessage = "Do not use Google to log in to a UC Davis account";
```

Comment:
No I would think it would work since it's declared in the layout page.  Maybe it's cleared because of multiple redirects somehow (since error message only stays around in TempData for the next request)?